src/encoding.ts

```diff
@@ -57,6 +59,11 @@ export function isAggregate(encoding: Encoding) {
   });
 }
 
+export function isRanged(encoding: Encoding) {
+  return (has(encoding, X) && has(encoding, X2)) ||
```

Comment:
I guess `(!!encoding.x  && !!encoding.x2) || (!!encoding.y  && !!encoding.y2)` should be sufficient.  (Or more specific  `(!!encoding.x.field || !!encoding.x.value) ...`)

Using `has` can also be incorrect since has only check `fieldDef.field`. 

If you specify `value` to `x` and `x2`, we should consider that "ranged" too.